redesign/generator_builder.py

```diff
@@ -1,29 +1,46 @@
 """Contains generator factory class that picks generator based on its profile type."""
 
-from .generators import GraphGenerator, TabularGenerator, UnstructuredGenerator
 from dataprofiler import StructuredProfiler, UnstructuredProfiler
 
+from .generators import GraphGenerator, TabularGenerator, UnstructuredGenerator
+
+
 class Generator:
     """Generator class."""
 
     valid_data_types = {
-        StructuredProfiler: TabularGenerator, 
-        UnstructuredProfiler: UnstructuredGenerator
+        StructuredProfiler: TabularGenerator,
+        UnstructuredProfiler: UnstructuredGenerator,
     }
 
     @classmethod
     def is_valid_data(cls, profile):
         return profile.__class__ in cls.valid_data_types
 
-    def __new__(cls, profile, config=None, options=None):
+    def __new__(cls, seed=None, config=None, *args, **kwargs):
         """Instantiate proper generator."""
         if config:
             try:
-                return config(profile, options)
+                return config(seed, *args, **kwargs)
```

Comment:
seed might not be the first argument, so maybe better to do `seed=seed` or put it in with the kwargs that would be a new dict